test/outpoint-test.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-env mocha */
+
+'use strict';
+
+const Outpoint = require('../lib/primitives/outpoint');
+const assert = require('./util/assert');
+const common = require('./util/common');
+const util = require('../lib/utils/util');
+const TX = require('../lib/primitives/tx');
+const OUTPOINT_SIZE = 36;
+
+const raw3 = common.readTX('tx2').getRaw();
+const raw4 = common.readTX('tx1').getRaw();
+console.log(TX.fromRaw(raw3).inputs);
+describe('Outpoint', () => {
+  let raw1, raw2, tx1, tx2, out1, out2;
+  beforeEach(() => {
+    tx1 = common.readTX('tx1').getRaw();
+    tx2 = common.readTX('tx3').getRaw();
+    raw1 = tx1.slice(5, 5+OUTPOINT_SIZE);
+    raw2 = tx2.slice(5, 5+OUTPOINT_SIZE);
+    out1 = Outpoint.fromRaw(raw1);
+    out2 = Outpoint.fromRaw(raw2);
+  });
+
+  it('should clone the outpoint correctly', () => {
+    const out1 = Outpoint.fromRaw(raw1);
+    const clone = out1.clone();
+    const equals = out1.equals(clone);
+
+    assert.strictEqual(out1 !== clone, true);
+    assert.strictEqual(equals, true);
+  });
+
+  it('should create outpoint from options object', () => {
+    const options = {};
+    options.hash = out1.hash;
+    options.index = out1.index;
+
+    const newOut = Outpoint.fromOptions(options);
+    assert(newOut.equals(out1), true);
+  });
+
+  it('should check hash and index are equal', () => {
+    const out1Clone = Outpoint.fromOptions(Object.assign(out1, {}));
+
+    assert(out1.equals(out2) === false, true);
+    assert(out1Clone.hash === out1.hash, out1Clone.equals(out1));
+    assert(out1Clone.index === out1.index, out1Clone.equals(out1));
+  });
+
+  it('should compare the indexes between outpoints', () => {
+    const index1 = out1.index;
+    const index2 = out2.index;
+    assert.equal(out1.compare(out2), index1 - index2);
```

Comment:
Even though this method is already tested in Inputs tests (BIP69). I guess it's okay to have specific one here. 

But `.compare` doesn't sort only by indexes. First it checks the TXID and sorts by TXID.. Only if TXIDs are equal it's sorted by indexes.

You can quickly Hack/Check by changing `out2.index = 2`, because TXIDs are different indexes aren't even compared.

Reply:
good catch!